java/drivers/driver-hazelcast4plus/src/main/java/com/hazelcast/simulator/tests/cp/CPMapTest.java

```diff
@@ -35,16 +45,23 @@ public class CPMapTest extends HazelcastTest {
     public int maps = 1;
     // number of distinct keys to create and use per-map; key domain is [0, keys)
     public int keys = 1;
+    // number of distinct value to create on every client (total number of values is valuesPerClient * clients)
+    public int valuesPerClient = 1;
     // size in bytes for each key's associated value
     public int valueSizeBytes = 100;
+
     private List<CPMap<Integer, String>> mapReferences;
 
-    private String v; // this is always the value associated with any key; exception is remove and delete
+    private String loadGeneratorValues[]; // this is always the value associated with any key; exception is remove and delete
 
+    private IList<CpMapOperationCounter> operationCounterList;
+    private ISet<String> allValues;
 
     @Setup
     public void setup() {
-        v = GeneratorUtils.generateAsciiString(valueSizeBytes);
+        loadGeneratorValues = createValues();
+        allValues = targetInstance.getSet(name + "AllValues");
+        allValues.addAll(Arrays.asList(loadGeneratorValues));
```

Comment:
I don't like this. 

If you want all load generator to share the same sequence of values, then make the creation of the values deterministic (e.g. ensuring that all load generators use the same seed).